src/routes/Home.jsx

```diff
@@ -25,7 +25,24 @@ const Home = () => {
     setSearchTags(newTags);
   };
 
-  const handleTagFilter = (e) => {};
+  const handleTagFilter = (e) => {
+    const et = e.target.innerText.slice(1);
+    if (et !== searchValue) {
+      setSearchValue(et);
+      const acc = [];
+      for (let p of posts) {
+        let chk = false;
+        for (let t of p.tags) {
+          if (t.content == et) chk = true;
+        }
+        if (chk) acc.push(p);
+      }
+      setPostList(acc);
+    } else {
+      setSearchValue("");
+      setPostList(posts);
+    }
```

Comment:
```suggestion
    const et = e.target.innerText.slice(1);
    if (et === searchValue) {
      setSearchValue("");
      setPostList(posts);
    } else {
      setSearchValue(et);
      const acc = posts.filter((post) =>
        post.tags.find((tag) => tag.content === et)
      );
      setPostList(acc);
    }
```
리팩토링 해보았습니다..! 참고 부탁드려용 😉